### ISSUE TITLE
ParamSpec argument passed Any should bound to Ellipsis

### ISSUE THREAD

**Issue by bijij** [microsoft/pyright#3176]
I have some code which accpts `Any` value, and if the value passed is `callable` it will call it with some pre-set arguments, (in the actual code-base, the signature is first checked before calling)

When a function expecting a callable, (typed via `ParamSpec`) is passed `Any`, the `args` and `kwargs` of the `ParamSpec` are left unbound, which causes type-checking to fail.

See this sample code:

```py
from typing import Any, Callable, TypeVar
from typing_extensions import ParamSpec

T = TypeVar('T')
P = ParamSpec('P')


def my_func(f: Callable[P, T], *args: P.args, **kwargs: P.kwargs) -> T:
    ...


def foo(a: int) -> None:
    if callable(a):
        reveal_type(a)
        reveal_type(my_func(a, 1))

def bar(a: Any) -> None:
    if callable(a):
        reveal_type(a)
        reveal_type(my_func(a, 1))

```

I would like if rather than being left unbound, the Callable to become either `Callable[..., Any]` or `Callable[..., Unknown]`

For comparison, I include the situation where pyright is aware the type is a callable, but has no other information, I believe they should behave similarly if not the same.

Thanks

-----

Sample screenshot:
![image](https://user-images.githubusercontent.com/8677174/158007070-b97d0913-4aa9-4254-b282-fc84a47c6cc1.png)

**Comment by erictraut** [comment]
Why are you using `Any` to annotate parameter `a` here? It's generally a bad idea to use an explicit `Any`, especially for parameters. Surely you must know something more about its type. For example, if you change it to `object`, it will type check without issue.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.230, which I just published. It will also be included in the next release of pylance.
